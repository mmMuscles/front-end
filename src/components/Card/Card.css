.card {
        padding: 20px;
        margin: 10px;
        box-shadow: 0 2px 8px rgba(0, 0, 0, 0.25);
        border-radius: 14px;
        background-color: rgba(255, 255, 255, 0.671);
}
