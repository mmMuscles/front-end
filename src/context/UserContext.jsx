import { createContext, useContext, useMemo, useState } from 'react';
import { supabase } from '../services/supabaseClient';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const session = supabase.auth.session()
  const [profile, setProfile] = useState({})
  const [user, setUser] = useState(
    session ? { id: session.user.id, email: session.user.email } : {}
  );
  
  const value = useMemo(() => ({ user, setUser, profile, setProfile}), [user, profile]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }
  return context;
};

export { UserProvider, useUser };
