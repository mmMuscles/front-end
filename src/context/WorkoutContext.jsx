import { useContext, createContext, useState } from "react";



const workoutCtx = createContext();


export const WorkoutProvider = ({ children }) => {
    const [workouts, setWorkouts] = useState([])

return <workoutCtx.Provider value={{workouts, setWorkouts}}>{children}</workoutCtx.Provider>
}



export const useWorkout = () => {
    const context = useContext(workoutCtx)

    if(context === undefined){
        throw new Error('Invalid Provider')
    }

    return context;
}
