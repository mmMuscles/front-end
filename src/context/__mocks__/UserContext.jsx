import { createContext, useContext, useMemo, useState } from "react";

const UserContext = createContext();

const UserProvider = ({ children, mockUser }) => {
  const [profile, setProfile] = useState({});
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const value = useMemo(
    () => ({ user, setUser, profile, setProfile }),
    [user, profile]
  );

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error("useUser must be used within a UserProvider");
  }
  return context;
};

export { UserProvider, useUser };
