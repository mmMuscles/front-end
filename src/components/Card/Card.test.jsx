import {render } from  '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { UserProvider } from '../../context/UserContext';
import Card from './Card';


describe('Header component test', () => {
  
    it('renders the top nav header', () => {
       const { container } = render(
            <UserProvider>
            <MemoryRouter>
                     <Card/>
            </MemoryRouter>
            </UserProvider>
        );
        expect(container).toMatchSnapshot();
         
    })

})
