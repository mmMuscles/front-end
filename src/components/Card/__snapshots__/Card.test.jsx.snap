// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders the top nav header 1`] = `
<div>
  <div
    class="card"
  />
</div>
`;
